extensions/extension_mgr.go

```diff
@@ -3,16 +3,48 @@
 package extensions
 
 import (
+	"embed"
 	"fmt"
 	"io"
+	"path"
 
+	"github.com/goccy/go-yaml"
 	substraitgo "github.com/substrait-io/substrait-go"
 	"github.com/substrait-io/substrait-go/proto/extensions"
-	"gopkg.in/yaml.v3"
 )
 
 type AdvancedExtension = extensions.AdvancedExtension
 
+const substraitDefaultURIPrefix = "https://github.com/substrait-io/substrait/blob/main/extensions/"
+
+// DefaultCollection is loaded with the default Substrait extension
+// definitions with the exception of decimal arithemtic. Decimal arithmetic
+// functions are missing as the complex return type expressions are not
+// yet implemented.
```

Comment:
Not sure if it'd help you, but CockroachDB has a Decimal implementation [here](https://github.com/cockroachdb/apd).

Reply:
Arrow has a full decimal implementation for decimal128 and decimal256 (that I wrote portions of haha) the issue isn't the decimal implementation but the [return type expression](https://substrait.io/expressions/scalar_functions/#return-type-expressions) parsing that I haven't implemented yet here. That file is the only one which uses the return type expressions, used to dynamically determine the precision and scale for the returned type.